src/GafferImage/Checkerboard.cpp

```diff
@@ -269,6 +269,10 @@ void Checkerboard::hashChannelData( const GafferImage::ImagePlug *output, const
 	h.append( channelName );
 
 	const int channelIndex = ImageAlgo::colorIndex( channelName );
+	if( channelIndex == -1 )
+	{
+		throw IECore::Exception( "Evaluated with invalid channel name: \"" + channelName + "\". This indicates a bug in a downstream node." );
+	}
```

Comment:
What's our intent here - to avoid a crash or to catch bad downstream clients? If the latter, then we should also be checking that the layer name is the one we expect.

Reply:
Lets see ... it's undefined behaviour to query a channel that doesn't exist ... other common nodes like Ramp don't check whether the channel name is valid, but they also don't crash. I was really just thinking about avoiding the crash, but I'm happy to add a check for the layer name if you want me to.